ca/certificate-authority.go

```diff
@@ -57,51 +59,104 @@ func NewCertificateAuthorityImpl(hostport string, authKey string, profile string
 	return
 }
 
-func (ca *CertificateAuthorityImpl) IssueCertificate(csr x509.CertificateRequest) (cert core.Certificate, err error) {
-	// XXX Take in authorizations and verify that union covers CSR?
+// When we receive CSRs from a client, we extract only the fields we care about
+// (names and public key), then create a new CSR from those to pass on to CFSSL.
+// The new CSR requires a signature, but that signature is not used for any
+// security purposes. The validity of the signing request is checked by the
+// registration authority, and the certificate authority authenticates itself to
+// CFSSL with CFSSL's own signature scheme. For the vestigial signature on the
+// re-wrapped CSR we use this single-purpose, insecure 512 bit "private" key.
+var csrResigningKey = `
+-----BEGIN RSA PRIVATE KEY-----
+MIIBOwIBAAJBAMLsxrThLEHt5Yzr4jxnn8rcX5HqhpkBKJ6V5gRAg+WXaJUR7mhe
+pAuZvNUbW2J127GvWH2QNkSNa/WMvOm/pH0CAwEAAQJAX2cb6jO7QZl6HHrnA8GE
+B/nMHNK4hfJ3OwcKyVH6PEgAB+7RaMqGsqdZ0vqabna6nM5rHYvPanuFgiFLKzqA
+AQIhAONw64nBvuPbqaA3GnDrOgzLsbZaz5Bz1FOxA5GxiWx9AiEA22agE9JsrRZR
+gjaZBfnTgNj1HDbPOrkr5Oui9AS8mAECIQCMHWadiSRGT27ias/5PJCYjWw/wRYa
+EaF+pZBjUxIsKQIhAMyHvD0eUi99edjA2yCGCBS6rK1zrvYYf4H15UBEVigBAiAH
+xilAU/SLRXqt06pbv/hshGAJhl3F8Rs1/8gf2CorAw==
+-----END RSA PRIVATE KEY-----
+`
+
+// Given a CSR provided by the user and validated by the registration authority,
+// extract the names and public key, and pass them on to a CFSSL instance for
+// signing. IMPORTANT: This method must only be called by the registration
+// authority, and only after verifying that all names are authorized for the
+// requesting account key.
+// TODO: Add technical controls to only accept messages from the registration
+// authority.
+func (ca *CertificateAuthorityImpl) IssueCertificate(unsafeCsr x509.CertificateRequest) (cert core.Certificate, err error) {
+	// Only sign RSA certificates until we explicitly support ECDSA.
+	if unsafeCsr.PublicKeyAlgorithm != x509.RSA {
+		err = errors.New("Non-RSA key provided in CSR.")
+	}
```

Comment:
Missing return after setting `err`. Also, this method looks long for naked returns: https://code.google.com/p/go-wiki/wiki/CodeReviewComments#Named_Result_Parameters